### PR TITLE
azure - new resources

### PR DESCRIPTION
**Pull request by stefangordon**
A batch of new resources.

```
policies:
  - name: frontdoor
    resource: azure.front-door
  - name: application-gateway
    resource: azure.application-gateway
  - name: logic-app-workflow
    resource: azure.logic-app-workflow
  - name: mysql
    resource: azure.mysql
  - name: service-fabric-cluster
    resource: azure.service-fabric-cluster
  - name: service-fabric-cluster-managed
    resource: azure.service-fabric-cluster-managed
  - name: traffic-manager-profile
    resource: azure.traffic-manager-profile
```

Also

- Cleanup soft delete resources in provisioning script
- Remove outdated/unnecessary Azure install instructions (the root instructions cover all clouds and these don't actually work due to setup.py not being frozen).

### CODE REVIEW

File: tools/c7n_azure/c7n_azure/resources/mysql.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+            filters:
+              - type: value
+                key: sku.name
+                op: equal
+                value: Basic
+
+    """
+
+    class resource_type(ArmResourceManager.resource_type):
+        doc_groups = ['Compute']
```
Comment (logachev) [inline tools/c7n_azure/c7n_azure/resources/mysql.py:30]: We have databases doc group: https://cloudcustodian.io/docs/azure/resources/group-databases.html

File: tools/c7n_azure/tests_azure/tests_resources/test_app_gateway.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+# SPDX-License-Identifier: Apache-2.0
+from ..azure_common import BaseTest
+
+
+class ApplicationGatewayTest(BaseTest):
+    def setUp(self):
+        super(ApplicationGatewayTest, self).setUp()
+
+    def test_app_gateway_validate(self):
+        with self.sign_out_patch():
```
Comment (logachev) [inline tools/c7n_azure/tests_azure/tests_resources/test_app_gateway.py:11]: nit: iirc, we don't really need this `with`.. Might be wrong, was a long time ago.

File: tools/c7n_azure/tests_azure/tests_resources/test_app_gateway.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
+# Copyright The Cloud Custodian Authors.
+# SPDX-License-Identifier: Apache-2.0
+from ..azure_common import BaseTest
+
+
+class ApplicationGatewayTest(BaseTest):
+    def setUp(self):
```
Comment (logachev) [inline tools/c7n_azure/tests_azure/tests_resources/test_app_gateway.py:7]: nit: I'd just remove empty methods (and other tests too)

**Review by logachev — APPROVED** [review]
lgtm, minor comments

File: tools/c7n_azure/c7n_azure/resources/mysql.py (hunk 1 above)
Comment (stefangordon) [reply tools/c7n_azure/c7n_azure/resources/mysql.py:30]: oh good catch

**Comment by stefangordon** [comment]
Closes #6718 #6717 #6716 #6712 #6711 #6710

**Pull request merged**